### PR TITLE
Spin up GRPC server when grpc-port is specified

### PR DESCRIPTION
**Pull request by lookuptable**

### CODE REVIEW

File: pkg/server/grpc/server.go [hunk 1]
```diff
@@ -53,7 +65,29 @@ func (s *server) HandleCSR(ctx context.Context, request *pb.Request) (*pb.Respon
...
+// Run starts a GRPC server on the specified port.
+func (s *Server) Run() error {
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", s.port))
+	if err != nil {
+		return fmt.Errorf("cannot listen on port %d (error: %v)", s.port, err)
+	}
+	grpcServer := grpc.NewServer()
+	pb.RegisterIstioCAServiceServer(grpcServer, s)
+
+	// grpcServer.Server() is a blocking call, so run it in a goroutine.
```
Comment (lizan) [inline pkg/server/grpc/server.go]: nit: grpcServer.Serve

File: pkg/server/grpc/server.go [hunk 2]
```diff
@@ -53,7 +65,29 @@ func (s *server) HandleCSR(ctx context.Context, request *pb.Request) (*pb.Respon
...
+	go func() {
+		glog.Infof("Starting GRPC server on port %d", s.port)
+
+		err := grpcServer.Serve(listener)
+
+		// grpcServer.Serve() always returns a non-nil error.
+		glog.Warningf("GRCP server returns an error: %v", err)
```
Comment (lizan) [inline pkg/server/grpc/server.go]: nit: s/GRCP/GRPC/

File: cmd/istio_ca/main.go [hunk 3]
```diff
@@ -117,6 +123,15 @@ func runCA() {
 	stopCh := make(chan struct{})
 	sc.Run(stopCh)
 
+	if opts.grpcPort > 0 {
+		grpcServer := grpc.New(ca, opts.grpcPort)
+		if err := grpcServer.Run(); err != nil {
+			glog.Warningf("GRPC server returns an error: %v", err)
```
Comment (wattli) [inline cmd/istio_ca/main.go]: What about "Fails to start the gRPC server with error: %v"

File: pkg/server/grpc/server.go [hunk 4]
```diff
@@ -24,11 +29,18 @@ import (
...
+type Server struct {
+	ca   ca.CertificateAuthority
+	port int
 }
 
-func (s *server) HandleCSR(ctx context.Context, request *pb.Request) (*pb.Response, error) {
+// HandleCSR handles an incoming certificate signing request (CSR). It does
+// proper validation (e.g. authentication) and upon validated, signs the CSR
+// and returns the resulting certificate. If not approved, reason for refusal
+// to sign is returnes as part of the response object.
```
Comment (wattli) [inline pkg/server/grpc/server.go]: s/returnes/returns

**Pull request merged**